<?xml version="1.0"?>

<launch>
    <arg name="hz" default="10"/>
    <arg name="robot_frame" default="base_link"/>
    <arg name="n_p" default="15"/>
    <arg name="n_h" default="3"/>
    <arg name="max_alpha" default="1.6"/>
    <arg name="max_psi" default="1.0"/>
    <arg name="n_s" default="1000"/>
    <arg name="target_velocity" default="0.8"/>
    <arg name="lookup_table_file_name" default="$(find state_lattice_planner)/lookup_table/lookup_table.csv"/>
    <arg name="max_iteration" default="5"/>
    <arg name="optimization_tolerance" default="0.1"/>
    <arg name="ignorable_obstacle_range" default="1.0"/>
    <arg name="verbose" default="false"/>
    <arg name="prediction_step" default="3.5"/>
    <arg name="collision_probability_threshold" default="0.05"/>
    <arg name="world_frame" default="map"/>

    <node pkg="slp_doa" type="slp_doa" name="slp_doa" ns="local_planner">
        <rosparam command="load" file="$(find state_lattice_planner)/config/robot_param.yaml" />
        <param name="HZ" value="$(arg hz)"/>
        <param name="ROBOT_FRAME" value="$(arg robot_frame)"/>
        <param name="N_P" value="$(arg n_p)"/>
        <param name="N_H" value="$(arg n_h)"/>
        <param name="MAX_ALPHA" value="$(arg max_alpha)"/>
        <param name="MAX_PSI" value="$(arg max_psi)"/>
        <param name="N_S" value="$(arg n_s)"/>
        <param name="TARGET_VELOCITY" value="$(arg target_velocity)"/>
        <param name="LOOKUP_TABLE_FILE_NAME" value="$(arg lookup_table_file_name)"/>
        <param name="MAX_ITERATION" value="$(arg max_iteration)"/>
        <param name="OPTIMIZATION_TOLERANCE" value="$(arg optimization_tolerance)"/>
        <param name="IGNORABLE_OBSTACLE_RANGE" value="$(arg ignorable_obstacle_range)"/>
        <param name="VERBOSE" value="$(arg verbose)"/>
        <param name="PREDICTION_STEP" value="$(arg prediction_step)"/>
        <param name="COLLISION_PROBABILITY_THRESHOLD" value="$(arg collision_probability_threshold)"/>
        <param name="WORLD_FRAME" value="$(arg world_frame)"/>
        <remap from="/local_map" to="/local_costmap"/>
        <remap from="/local_goal" to="/local_goal"/>
        <remap from="/cmd_vel" to="/velocity"/>
        <remap from="/odom" to="/odom"/>
    </node>
</launch>
